cdap-api/src/main/java/co/cask/cdap/api/schedule/TriggerFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import co.cask.cdap.api.ProgramStatus;
+import co.cask.cdap.api.app.ProgramType;
+import co.cask.cdap.internal.schedule.ScheduleCreationSpec;
+
+/**
+ * A factory for getting a specific type of {@link Trigger}
+ */
+public interface TriggerFactory {
+
+  Trigger or(Trigger... triggers);
+
+  Trigger and(Trigger... triggers);
+
+  /**
+   * Creates a schedule which is triggered based upon the given cron expression.
+   *
+   * @param cronExpression the cron expression to specify the time to trigger the schedule
+   * @return a {@link ScheduleCreationSpec}
+   */
+  Trigger byTime(String cronExpression);
```

Comment:
Can this to be renamed to `onTime`? Isn't it technically a new method and won't deprecate any APIs? Or was there another reason this is `byTime`?